Replace confd

hparfr opened the pull request:
alternative to #18 
resolves #17

hparfr commented on bin/entrypoint.sh:
```diff
@@ -8,14 +8,7 @@ USER_ID=${LOCAL_USER_ID:-999}
-export PGUSER=${DB_USER}
-export PGPASSWORD=${DB_PASSWORD}
-export PGDATABASE=${DB_NAME}
+source /odoo/templates/answers.sh && envsubst < /odoo/templates/odoo.cfg.tmpl > /etc/odoo.cfg
```
@sbidoul  confd replacement is here

sbidoul replied:
Yes, feel free to continue this route :)

sbidoul commented on templates/14.0/odoo.cfg.tmpl:
```diff
@@ -0,0 +1,50 @@
+sentry_traces_sample_rate = ${SENTRY_TRACES_SAMPLE_RATE}
+encryption_key_dev=${ENCRYPTION_KEY_DEV}
+encryption_key_ci=${ENCRYPTION_KEY_CI}
+encryption_key_prod=${ENCRYPTION_KEY_PROD}
```
Let's think about new environment variables in another PR.

hparfr replied:
yes sure, it's was an example from another project.

sbidoul commented:
@hparfr I took some time to add a test suite that validates many of the promises of this image, including the expected /etc/odoo.cfg built from environment variables.

Could you rebase to check how it goes?

sbidoul commented on Dockerfile-14.0:
```diff
@@ -16,7 +16,6 @@ COPY ./install /tmp/install
 RUN set -x \
   && /tmp/install/pre-install.sh \
   && /tmp/install/tools.sh \
-  && /tmp/install/confd.sh \
```
Don't forget to remove `install/confd.sh` too.
And to update the other Dockerfiles.

hparfr commented:
@sbidoul 

Thanks for adding tests.

I have reduced the number of commits, removed confd install file and references in dockerfiles,  fixed default values and moved PG_ definition from entrypoint to answers.sh

The PR is ready.

sbidoul commented:
Thank!

sbidoul commented on templates/10.0/answers.sh:
```diff
@@ -0,0 +1,33 @@
+declare -x DB_NAME="${DB_NAME:-}"
+declare -x DB_USER="${DB_USER:-}"
+declare -x DB_PASSWORD="${DB_PASSWORD:-}"
+declare -x DB_FILTER="${DB_FILTER:-^(odoodb)$}"
```
@hparfr `odoodb` iznogoud, here :)
